openedx/core/djangoapps/programs/utils.py

```diff
@@ -136,19 +138,31 @@ def progress(self, programs=None, count_only=True):
                 progress, and unstarted courses instead of serialized representations
                 of the courses.
 
+            show_expired_runs_as_remaining (bool): Whether in progress runs whose upgrade
```

Comment:
We don't want this method's output to change across the list and detail views. The user's view of progress should be consistent across those pages. This should just be how the method behaves, not an option.

We're also at risk of introducing debt here. It's not safe to assume that just because the upgrade deadline is past, completion of the course run will no longer count for completion of the course. Each program has a type, and those types define a [set of seat types](https://github.com/edx/course-discovery/blob/master/course_discovery/apps/course_metadata/models.py#L649) that qualify for completion of the program. Given that we have support for that, we shouldn't be hardcoding assumptions that all programs require verified seats like you're doing here.

At the very least, this should behave like the ProgramProgressMeter's `completed_programs` property. Read [this](https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/programs/utils.py#L207-L213).